From 59486ced2e338456d13b4c33b04943097eba9fba Mon Sep 17 00:00:00 2001
From: chalesyu <574249312@qq.com>
Date: Sat, 13 Feb 2021 00:00:00 +0800
Subject: [PATCH 2/6] dt-bindings:sunxi:h6:Fix UART over SD variant

include
	led and usb power pins
	disable mmc0(sdcard) node
	currect uart0 pins

dt-bindings:sunxi:h6:Fix EMMC for txh6
---
 arch/arm/dts/sun50i-h6-tanix-tx6.dts | 51 ++++++++++++++++++++++++++++
 arch/arm/dts/sun50i-h6.dtsi          | 10 +++---
 2 files changed, 57 insertions(+), 4 deletions(-)

diff --git a/arch/arm/dts/sun50i-h6-tanix-tx6.dts b/arch/arm/dts/sun50i-h6-tanix-tx6.dts
index be81330db1..537542e275 100644
--- a/arch/arm/dts/sun50i-h6-tanix-tx6.dts
+++ b/arch/arm/dts/sun50i-h6-tanix-tx6.dts
@@ -32,6 +32,29 @@
 		};
 	};
 
+	leds {
+		compatible = "gpio-leds";
+
+		power {
+			label = "tx6:white:power";
+			gpios = <&r_pio 0 4 GPIO_ACTIVE_HIGH>; /* PL4 */
+			default-state = "on";
+		};
+	};
+
+	reg_vcc1v8: vcc1v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc1v8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+	};
+
 	reg_vcc3v3: vcc3v3 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc3v3";
@@ -72,6 +95,20 @@
 	status = "okay";
 };
 
+&uart0 {
+	status = "okay";
+};
+
+&dwc3 {
+	status = "okay";
+};
+
+&emac {
+	phy-mode = "rmii";
+	//phy-handle = <&ext_rmii_phy>;
+	status = "okay";
+};
+
 &hdmi {
 	status = "okay";
 };
@@ -88,6 +125,16 @@
 	vmmc-supply = <&reg_vcc3v3>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;
 	bus-width = <4>;
+	status = "disabled";
+};
+
+
+&mmc2 {
+	vmmc-supply = <&reg_vcc3v3>;
+	vqmmc-supply = <&reg_vcc1v8>;
+	cap-mmc-hw-reset;
+	non-removable;
+	bus-width = <4>;
 	status = "okay";
 };
 
@@ -122,3 +169,7 @@
 &usb3phy {
 	status = "okay";
 };
+
+&usb3phy {
+	status = "okay";
+};
diff --git a/arch/arm/dts/sun50i-h6.dtsi b/arch/arm/dts/sun50i-h6.dtsi
index 8a62a9fbe3..45f26a2ee2 100644
--- a/arch/arm/dts/sun50i-h6.dtsi
+++ b/arch/arm/dts/sun50i-h6.dtsi
@@ -334,8 +334,8 @@
 			};
 
 			mmc0_pins: mmc0-pins {
-				pins = "PF0", "PF1", "PF2", "PF3",
-				       "PF4", "PF5";
+				pins = "PF0", "PF1", "PH0", "PF3",
+				       "PH1", "PF5";
 				function = "mmc0";
 				drive-strength = <30>;
 				bias-pull-up;
@@ -390,7 +390,7 @@
 			};
 
 			uart0_ph_pins: uart0-ph-pins {
-				pins = "PH0", "PH1";
+				pins = "PF2", "PF4";
 				function = "uart0";
 			};
 
@@ -481,7 +481,9 @@
 			reg-io-width = <4>;
 			clocks = <&ccu CLK_BUS_UART0>;
 			resets = <&ccu RST_BUS_UART0>;
-			status = "disabled";
+			status = "okay";
+			pinctrl-names = "default";
+			pinctrl-0 = <&uart0_ph_pins>;
 		};
 
 		uart1: serial@5000400 {
-- 

